src/instructlab/sdg/datamixing.py

```diff
@@ -362,17 +362,28 @@ def __pick_documents(rec, p):
     return ds
 
 
-def _conv_pretrain(rec):
+def _conv_pretrain(rec, student_model_arch: str):
```

Comment:
This is fine in the CLI, but i would actually caution against having any hard ties to model architecture in SDG. Since the system prompt is already being passed in separately, SDG doesn't need to know about specific model architectures. Instead, it should probably just take in a boolean like `use_legacy_tokens`, that the CLI can set as true/false depending on the arch

Reply:
So like in the CLI at the same conditional where you pick which system prompt to pass in, you could also just choose whether to set this to true false

Reply:
updated, thanks